hello.py

```diff
@@ -870,22 +892,41 @@ def main():
         registry2=registry2,
         snapshotter=snapshotter,
         cleanup=cleanup,
+        insecure_registry=insecure_registry,
     )
 
+    if output_format == "csv":
+        csv_headers = (
+            "timestamp,repo,bench,pull_time(s),create_time(s),run_time(s),elapsed(s)"
+        )
+        f.write(csv_headers + "\n")
+        f.flush()
+
     for bench in benches:
         pull_time, create_time, run_time = runner.operation(op, bench)
 
-        row = {
-            "repo": bench.repo,
-            "bench": bench.name,
-            "pull_time": f"{pull_time: .6f}",
-            "create_time": f"{create_time: .6f}",
-            "run_time": f"{run_time: .6f}",
-            "elapsed": f"{pull_time + create_time + run_time: .6f}",
-        }
-        js = json.dumps(row)
-        print(js)
-        f.write(js + "\n")
+        elapsed_time = f"{pull_time + create_time + run_time: .6f}"
+        timetamp = int(time.time() * 1000)
+        pull_time = f"{pull_time: .6f}"
+        cleate_time = f"{create_time: .6f}"
+        run_time = f"{run_time: .6f}"
+
+        if output_format == "json":
+            row = {
+                "timestamp": timetamp,
+                "repo": bench.repo,
+                "bench": bench.name,
+                "pull_time": pull_time,
+                "create_time": cleate_time,
+                "run_time": run_time,
+                "elapsed": elapsed_time,
+            }
+            line = json.dumps(row)
+        elif output_format == "csv":
+            line = f"{timetamp},{bench.repo},{bench.name},{pull_time},{cleate_time},{run_time},{elapsed_time}"
+
+        print(line)
+        f.write(line + "\n")
```

Comment:
Try `writelines`

Reply:
Fixed. I did not make a design to write once. Because all datas will be lost when one task fails.